File: client/src/pages/trip.js
import { useLocation, useParams } from "react-router-dom";

function TripPage() {
  const { tripIndex } = useParams();
  const location = useLocation();
  const { trip } = location.state || {};

  if (!trip) {
    return <p>No trip data found. Try navigating from the sidebar.</p>;
  }

  return (
    <div>
      <h1>
        Trip #{tripIndex}: {trip.source} → {trip.destination}
      </h1>

      <p>
        {trip.days} days, {trip.people_count} people, budget ${trip.budget}
      </p>

      {trip.flights && trip.flights.length > 0 && (
        <>
          <h2>Flights</h2>
          <ul>
            {trip.flights.map((f, i) => (
              <li key={i}>
                {f.airline} via {f.provider} - ${f.price} ({f.duration_hours}h)
              </li>
            ))}
          </ul>
        </>
      )}

      {trip.hotels && trip.hotels.length > 0 && (
        <>
          <h2>Hotels</h2>
          <ul>
            {trip.hotels.map((h, i) => (
              <li key={i}>
                {h.name} via {h.provider} - ${h.price_per_day}/night ({h.rating}★)
              </li>
            ))}
          </ul>
        </>
      )}

      {trip.itinerary && trip.itinerary.length > 0 && (
        <>
          <h2>Itinerary</h2>
          <ul>
            {trip.itinerary.map((item, i) => (
              <li key={i}>
                Day {item.day}: {item.plan}
              </li>
            ))}
          </ul>
        </>
      )}

      <h2>Raw plan JSON</h2>
      <pre>{JSON.stringify(trip, null, 2)}</pre>
    </div>
  );
}

export default TripPage;

File: server/app/schema/trip.py
from typing import List
from pydantic import BaseModel, Field

class FlightOption(BaseModel):
    provider: str
    airline: str
    price: float
    duration_hours: int

class HotelOption(BaseModel):
    provider: str
    name: str
    price_per_day: float
    rating: float

class ItineraryItem(BaseModel):
    day: int
    plan: str

class TripPlanRequest(BaseModel):
    source: str = Field(..., min_length=1)
    destination: str = Field(..., min_length=1)
    people_count: int = Field(..., gt=0)
    days: int = Field(..., gt=0)
    budget: float = Field(..., gt=0)

class TripPlanResponse(BaseModel):
    source: str
    destination: str
    days: int
    people_count: int
    budget: float
    flights: List[FlightOption]
    hotels: List[HotelOption]
    itinerary: List[ItineraryItem]